app/controllers/assignments_controller.rb

```diff
@@ -832,7 +832,22 @@ def process_assignment_form(assignment)
       periods = period_attrs.to_h.values.map { |h| h[:id].presence }
       assignment.submission_rule.periods.where.not(id: periods).find_each(&:destroy)
     end
-    assignment.assign_attributes(assignment_params)
+
+    # Handle "scheduled" visibility option
+    # When is_hidden="scheduled", convert to is_hidden=false with datetime values
+    params_copy = assignment_params.deep_dup
+    if params_copy[:is_hidden] == 'scheduled'
+      params_copy[:is_hidden] = false
+    end
+
+    # Handle section-specific "scheduled" visibility
+    params_copy[:assessment_section_properties_attributes]&.each_value do |section_props|
```

Comment:
In addition to the comment I left above, there is another bug that I found when testing: when the user first selects "Visible during scheduled time" (with a visible from/visible until), and then later switches back to "Hidden", "Visible", or (for sections-specific settings) "Default", _the visible from/visible until fields should be set to nil_.